### PR TITLE
fix: enhance session comparison (SDK V2)

### PR DESCRIPTION
**Pull request by ffmcgee725**
## Explanation

When we are managing session upgrade in both transports, we currently check is `sessionScopes` have changes but we should also be checking if the accounts are the same ones as well, and if not, upgrade the session.

This PR creates an enhanced scope checking utility function to be used in both transports.

## References

* Fixes https://www.notion.so/metamask-consensys/SDK-Phase-1-End-game-276f86d67d688061a461ee0cbaf37649?p=281f86d67d688077aab0eb8637e82dcb&pm=s

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

---

> [!NOTE]
> Adds `isSameScopesAndAccounts` and updates transports to recreate sessions when scopes or proposed accounts differ; includes comprehensive tests.
> 
> - **Utils**:
>   - Add `isSameScopesAndAccounts(currentScopes, proposedScopes, walletSession, proposedCaipAccountIds)` to validate both scopes and CAIP accounts against existing `sessionScopes`.
> - **Transports**:
>   - `multichain/transports/default` and `multichain/transports/mwp` now use `isSameScopesAndAccounts` when connecting/resuming; recreate (create + revoke) session if mismatch.
> - **Tests**:
>   - Add extensive unit tests for `isSameScopesAndAccounts`, covering exact matches, mismatched scopes, subset accounts, empty/absent accounts, ordering differences, and multiple accounts per scope.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 21a6ce2e3c0033718a2fd62cbb30103c6bcde392. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/sdk-multichain/src/multichain/utils/index.ts [hunk 1]
```diff
@@ -141,6 +141,42 @@ export function setupDappMetadata(options: MultichainOptions): MultichainOptions
...
+  if (!isSameScopes) {
+    return false;
+  }
+
+  const existingAccountIds: CaipAccountId[] = [];
+  for (const scopeObject of Object.values(walletSession.sessionScopes)) {
+    if (scopeObject.accounts) {
+      existingAccountIds.push(...scopeObject.accounts);
+    }
+  }
```
Comment (wenfix) [inline packages/sdk-multichain/src/multichain/utils/index.ts]:
```suggestion
  const existingAccountIds: CaipAccountId[] = Object.values(walletSession.sessionScopes)
  .filter(({accounts}) => Boolean(accounts))
  .flatMap(({accounts}) => accounts)
```

Comment (ffmcgee725) [reply packages/sdk-multichain/src/multichain/utils/index.ts]: https://github.com/MetaMask/metamask-sdk/pull/1365/commits/21a6ce2e3c0033718a2fd62cbb30103c6bcde392
